ablySpec/RealtimeClientChannel.swift

```diff
@@ -186,6 +187,56 @@ class RealtimeClientChannel: QuickSpec {
             // RTL6
             describe("publish") {
 
+                // RTL6a
+                it("should encode messages in the same way as the RestChannel") {
+                    let data = ["value":1]
+                    var restEncodedMessage: ARTMessage?
+                    var realtimeEncodedMessage: ARTMessage?
+
+                    let rest = ARTRest(options: AblyTests.commonAppSetup())
+                    let restChannel = rest.channels.get("test")
+
+                    let restBlock: @convention(block) (AspectInfo, ARTMessage) -> Void = { info, message in
+                        let invocation = info.originalInvocation()
+                        invocation.invoke()
```

Comment:
`invocation.invoke()` calls the original implementation, so the `encodeWithEncoder` will write to the current `ARTMessage` instance (look https://github.com/ably/ably-ios/blob/bff8f901b67e9ab12b38fa477470ed773ae0a902/ably-ios/ARTChannel.m#L54). Even so, I saw the documentation and I think it is possible to retrieve the original return value of the mehotd directly with the [NSInvocation.getReturnValue](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Classes/NSInvocation_Class/#//apple_ref/occ/instm/NSInvocation/getReturnValue:). I think both approaches are correct. What you think?